docs/USER_RECORD.md

```diff
@@ -0,0 +1,932 @@
+
+5. Various fields are located in the `status` section (a sub-sub-object of the
+   user record, also with an intermediary object between that is keyed by the
+   machine ID, similar to the way the `binding` section is organized). This
+   section is augmented during runtime only, and never persisted to disk. The
+   idea is that this section contains information about current runtime
+   resource usage (for example: currently used disk space of the user), that
+   changes dynamically but is otherwise immediately associated with the user
+   record and for many purposes should be considered to be part of the user
+   record.
+
+6. The `signature` section contains one or more cryptographic signatures of a
+   reduced version of the user record. This is used to ensure that only user
+   records defined by a specific source are accepted on a system, by validating
+   the signature against the set of locally accepted signature public keys. The
+   signature is calculated from the JSON user record with all sections removed,
+   except for `regular`, `privileged`, `perMachine`. Specifically, `binding`,
+   `status`, `signature` itself and `secret` are removed first and thus not
+   covered by the signature. This section is optional, and is only used when
+   cryptographic validation of user records is required (as it is by
+   `systemd-homed.service` for example).
+
+7. The `secret` section contains secret user credentials, such as password or
+   PIN information. This data is never persisted, and never returned when user
+   records are inquired by a client, privileged or not. This data should only
+   be included in a user record very briefly, for example when certain very
+   specific operations are executed. For example, in tools such as
+   `systemd-homed` this section may be included in user records, when creating
+   a new home directory, as passwords and similar credentials need to be
+   provided to encrypt the home directory with.
+
+Here's a tabular overview of the sections and their properties:
+
+| Section    | Included in Signature | Persisted | Security Sensitive | Contains Host-Specific Data |
+|------------|-----------------------|-----------|--------------------|-----------------------------|
+| regular    | yes                   | yes       | no                 | no                          |
+| privileged | yes                   | yes       | yes                | no                          |
+| perMachine | yes                   | yes       | no                 | yes                         |
+| binding    | no                    | yes       | no                 | yes                         |
+| status     | no                    | no        | no                 | yes                         |
+| signature  | no                    | yes       | no                 | no                          |
+| secret     | no                    | no        | yes                | no                          |
```

Comment:
well, if these user records are used by homed, then yes, the user record stored on the host will contain the binding, but the two copies in the user's home directory + LUKS2 header will not contain it. That said, the user records are supposed to be suitable for homed but not exclusive to it, hence I shied away from documenting here how this is done in homed nad that it has to be done that way, because it doesn't really. people can decide never to use "binding" actually and stick everything into the regular section, and that's entirely fine.  

Anyway will add a comment about the application of this to homed here, maybe it helps.